alicloud/resource_alicloud_snapshot_test.go

```diff
@@ -0,0 +1,300 @@
+		}
+	}
+
+	if sweeped {
+		time.Sleep(30 * time.Second)
+	}
+	return nil
+}
+
+func TestAccAlicloudSnapshot_basic(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			testAccPreCheck(t)
+		},
+
+		// module name
+		IDRefreshName: "alicloud_snapshot.snapshot",
+
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckSnapshotDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccSnapshotConfig,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckSnapshotExists("alicloud_snapshot.snapshot"),
+					resource.TestCheckResourceAttrSet("alicloud_snapshot.snapshot", "disk_id"),
+					resource.TestCheckResourceAttr("alicloud_snapshot.snapshot", "name", "tf-testAcc-snapshot"),
+					resource.TestCheckResourceAttr("alicloud_snapshot.snapshot", "description", "TF Test"),
+					resource.TestCheckResourceAttr("alicloud_snapshot.snapshot", "tags.%", "1"),
+					resource.TestCheckResourceAttr("alicloud_snapshot.snapshot", "tags.version", "1.0"),
+				),
+			},
+			resource.TestStep{
+				Config: testAccSnapshotConfigUpdate,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckSnapshotExists("alicloud_snapshot.snapshot"),
+					resource.TestCheckResourceAttrSet("alicloud_snapshot.snapshot", "disk_id"),
+					resource.TestCheckResourceAttr("alicloud_snapshot.snapshot", "name", "tf-testAcc-snapshot"),
+					resource.TestCheckResourceAttr("alicloud_snapshot.snapshot", "description", "TF Test"),
+					resource.TestCheckResourceAttr("alicloud_snapshot.snapshot", "tags.%", "1"),
+					resource.TestCheckResourceAttr("alicloud_snapshot.snapshot", "tags.version", "1.1"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckSnapshotDestroy(s *terraform.State) error {
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "alicloud_snapshot" {
+			continue
+		}
+
+		client := testAccProvider.Meta().(*connectivity.AliyunClient)
+		ecsService := EcsService{client}
+
+		_, err := ecsService.DescribeSnapshotById(rs.Primary.ID)
+
+		if err != nil {
+			if NotFoundError(err) {
+				continue
+			}
+			return fmt.Errorf("Describing snapshot (%s) failed while destoring, error: %#v.", rs.Primary.ID, err)
+		}
+		return fmt.Errorf("Error ECS Snapshot (%s) still exist", rs.Primary.ID)
+	}
+
+	return nil
+}
+
+func testAccCheckSnapshotExists(n string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("Not found: %s", n)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No Snapshot ID is set")
+		}
+
+		client := testAccProvider.Meta().(*connectivity.AliyunClient)
+		ecsService := EcsService{client}
+		_, err := ecsService.DescribeSnapshotById(rs.Primary.ID)
+
+		if err != nil {
+			return fmt.Errorf("While checking disk existing, describing disk got an error: %#v.", err)
+		}
+
+		return nil
+	}
+}
+
+const testAccSnapshotConfig = `
+data "alicloud_instance_types" "instance_type" {
+}
```

Comment:
The alicloud_instance_types should add availability_zone to get the specified instance types.